rfd/0124-plugin-secrets.md

```diff
@@ -0,0 +1,161 @@
+---
+authors: Michael Wilson (michael.wilson@goteleport.com)
+state: draft
+---
+
+# RFD 124 - Plugin static credentials
+
+### Required Approvers
+
+* Engineering @r0mant, @justinas
+* Security @reed
+* Product: @klizhentas
+
+## What
+
+Allow storage of plugin static credentials within Teleport in a safe and secure way.
+
+## Why
+
+As we grow our plugin support, we have discovered that the various services we
+hope to integrate do not offer a uniform mechanism for configuration. For instance,
+Slack uses oauth2, but Okta requires an API token, and Jamf requires a username and
+password.
+
+Today, the plugin houses a credentials object that allows for this sort of storage.
+However, unlike an oauth authorization code, static credentials pose a much greater risk of
+exposure, and rotating credentials can be a time consuming process.  To reduce the risks associated
+with storing static credentials within Teleport and to allow for easier credentials rotation,
+we would like to create a new resource within Teleport that is handled in an explicitly secure
+way.
+
+## Details
+
+### The `PluginStaticCredentials` object
+
+A new object will be needed to house static credentials. At present it will support:
+
+- API tokens
+- Usernames and passwords
+- OAuth client ID and secret
+
+This object has very strict access control requirements that will be explained in detail
+in the **Access Control** section.
+
+```yaml
+kind: plugin_static_credentials
+version: v1
+metadata:
+  name: plugin-name # name must match the name of the plugin.
+spec:
+  last_rotated: "2023-05-09T16:50:52.497874Z"
```

Comment:
How is this filled in? Is it updated on every update of the resource?

Reply:
Ahh, I'm punting on the rotation for now, so this is something that we can figure out later. But yes, that's what I was thinking.